telegram/_chatmember.py

```diff
@@ -356,6 +356,7 @@ class ChatMemberRestricted(ChatMember):
     .. versionchanged:: 20.0
        The argument :paramref:`can_manage_topics` was added, which changes the position of the
        optional argument :paramref:`until_date`.
+       (Edit: This was a mistake, the argument is of course not optional)
```

Comment:
until_date is not an optional argument. It confused me so I added the note, I imagine a user could be confused as well

Reply:
The note confused me more than the original statement 🙈  Maybe just remove the "optional"? 🤔

Reply:
But then I would ask why in .0 it changed the position and in .1 it didn't.

I mean, I guess no one else will probably notice this, I could also just drop it.

Reply:
Ah okay. Then maybe just add "Moreover `until_date` was converted into a positional argument to comply with the Bot API."?

Reply:
Hm, it wasn't converted. 

The split into different ChatMember classes way back 13.7 or so changed until_date into a required argument. That was done correctly. 

When can_manage_topics came around with V20, this note was added incorrectly and, also incorrectly, can_manage_topics was added before until_date as another required argument. Because for some reason we thought when doing this that until_date was still optional. Which it is not.

So the note is wrong, that decision/the position of can_manage_topics is wrong, but that is it. I just wanted to point out that the note is wrong, and I added all the new media permissions after until_date.

Reply:
> Hm, it wasn't converted.

[here](https://docs.python-telegram-bot.org/en/v13.11/telegram.chatmemberrestricted.html#telegram.ChatMemberRestricted) it's optional. [here](https://docs.python-telegram-bot.org/en/stable/telegram.chatmemberrestricted.html#telegram.ChatMemberRestricted) it's not …

> The split into different ChatMember classes way back 13.7 or so changed until_date into a required argument. That was done correctly.
> 
> When can_manage_topics came around with V20, this note was added incorrectly and, also incorrectly, can_manage_topics was added before until_date as another required argument. Because for some reason we thought when doing this that until_date was still optional. Which it is not.
> 
> So the note is wrong, that decision/the position of can_manage_topics is wrong, but that is it. I just wanted to point out that the note is wrong, and I added all the new media permissions after until_date.

Sorry, I just don't get what you're trying to say 😭

in v13.11 (base of v20), until_date` is optional (which is wrong) and there is no `can_manage_topics`. in v20, all arguments are positional and there is `can_manage_topics` - before `until_date`. So pretty much everything is messaged up. Right?

Reply:
Yes, agreed. So what do we do about it?

I would honestly be fine with just dropping my edit to the note and assume no one reads too closely into it. Or we replace it completely with something like

"in V20 we messed up adding can_manage_topics to the correct position"

without explaining why because it doesn't matter and is just confusing.

Reply:
"All arguments were made positional and their order was changed. The argument `can_manage_topics` was added."

nice formulation of "everything went nuts"